### ISSUE TITLE
1.0Ha sub-release not supporting multiple mappings

### ISSUE THREAD

**Issue by IohannesIohannium** [Idhrendur/paradoxGameConverters#507]
The 1.0Ha sub-release doesn't correctly read the country mappings and, in instances where they would be more than one country mapped, only the last one is read.

E.g.:
```
link = { eu4 = TUR v2 = TUR } # Ottomans
link = { eu4 = k_ottoman v2 = TUR } # Ottomans
link = { eu4 = k_anatolia v2 = TUR } # Anatolia => Turkey
```

The converter will completely discard the first two lines and generate a new country for the EU4 "TUR", assigning them a random flag, as same thing happens also, I guess, with CK2 name-to-tile file.

**Comment by IohannesIohannium** [comment]
Have I to give a proof, here I upload the savefile, along with the mod and the log generated.

[mod.zip](https://github.com/Idhrendur/paradoxGameConverters/files/1822965/mod.zip)

Furtherly going on, upon loading it into Vic2, the game refuses to let me select any country, in both bookmarks (1821 and 1836). It also seems to have re-populated Africa (I cleansed it in the EU4 savefile).

**Comment by kratostatic** [comment]
Right.  So the second problem (which I don't get myself but others have) can be fixed just by reversing (or unreversing rather) the order of the bookmarks in the bookmark file in the mod.  I'm not quite sure what to do with 1821, as I don't want to have it as the "default" if things aren't working with it but I do want to have one.  I seem to remember there being an 1821 mod for Vicky 2, I might look into how that did it and/or see if the author is still about to ask if we can just take bits.

The first is more complicated, I think it might be an issue with the new parser, but I'll be looking into it anyway.  Thanks for digging into the issue, there have been a couple of reports of flag oddness and I had no idea where to start looking!

**Comment by Idhrendur** [comment]
I'm a bit confounded by the country mapping thing, but it should be easy enough to check they're getting read right (just drop a breakpoint at the end of mappers::CountryMappings::readRules() and see what made it into the list. Looking at the code, it *should* work (but if testing shows otherwise, it clearly doesn't).

Do note that you now need to make sure EU4 is capitalized in a mapping, and you need to use Vic2 (with the v capitalized) instead of v2.
